modules/javafx.graphics/src/main/java/com/sun/glass/utils/NativeLibLoader.java

```diff
@@ -301,7 +303,22 @@ private static String cacheLibrary(InputStream is, String name, Class caller) th
         }
         if (write) {
             Path path = f.toPath();
-            Files.copy(is, path);
+            File lockFile = new File(cacheDir, ".lock");
+            try (RandomAccessFile randomAccessFile = new RandomAccessFile(lockFile, "rw");
+                 FileChannel fileChannel = randomAccessFile.getChannel();
+                 FileLock fileLock = fileChannel.lock()) {
+                try {
+                    if (!Files.exists(path)) {
```

Comment:
This is a different type of a problem, that imho should be addressed as well, but in a separate issue which also contains a test strategy. This is something that requires a hash-based solution, but it is something that needs to be done with great care, so I personally wouldn't rush this into this PR (as it might cause criticial regression if not done correctly).

Reply:
Agree to do it separately, but in any case, if there was already a file at the cache directory, its checksum is verified, and in case it is not the same, the file is removed from the cache, before it is copied again.
Does this cover the possible scenarios you are referring to?

Reply:
I agree with Johan and Jose: This should be fixed in a separate issue. @jperedadnr can you file a follow-up bug to investigate?

Reply:
Sure, I've filed: https://bugs.openjdk.org/browse/JDK-8313688